tsdb/index/tsi1/partition.go

```diff
@@ -909,6 +952,30 @@ func (p *Partition) compact() {
 	fs := p.retainFileSet()
 	defer fs.Release()
 
+	// compact any non-active log files first
+	for _, f := range p.fileSet.files {
+		if f.Level() == 0 {
+			logFile := f.(*LogFile) // It is an invariant that a file is level 0 iff it is a log file
+			if logFile == p.activeLogFile {
+				continue
+			}
+			if p.levelCompacting[0] {
+				break
+			}
+			// Mark the level as compacting.
+			p.levelCompacting[0] = true
```

Comment:
Do we need to take a lock here?

Reply:
We have already taken it:

```
// Compact requests a compaction of log files.
func (p *Partition) Compact() {
	p.mu.Lock()
	defer p.mu.Unlock()
	p.compact()
}
```

Reply:
I'd like a comment at the start of little 'c' `compact()` saying that this assumes the lock is taken.